crates/biome_rule_options/src/use_consistent_arrow_return.rs

```diff
@@ -1,6 +1,45 @@
+use biome_deserialize::{
+    DeserializableValidator, DeserializationContext, DeserializationDiagnostic,
+};
 use biome_deserialize_macros::Deserializable;
 use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Deserializable, Eq, PartialEq, Serialize, Default)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase")]
+pub enum UseConsistentArrowReturnStyle {
+    #[default]
+    AsNeeded,
+    Always,
+    Never,
+}
+
 #[derive(Default, Clone, Debug, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
 #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[serde(rename_all = "camelCase", deny_unknown_fields, default)]
-pub struct UseConsistentArrowReturnOptions {}
+#[deserializable(with_validator)]
+pub struct UseConsistentArrowReturnOptions {
+    pub style: UseConsistentArrowReturnStyle,
+    pub require_for_object_literal: bool,
+}
+
+impl DeserializableValidator for UseConsistentArrowReturnOptions {
+    fn validate(
+        &mut self,
+        ctx: &mut impl DeserializationContext,
+        _name: &str,
+        range: biome_rowan::TextRange,
+    ) -> bool {
+        if self.require_for_object_literal && self.style != UseConsistentArrowReturnStyle::AsNeeded
+        {
+            ctx.report(
+                DeserializationDiagnostic::new(
+                    "`require_for_object_literal` can only be used when `style` is `asNeeded`.",
```

Comment:
This diagnostic is presented to users, so it must refer to option names as they are seen by the user. In this case, `requireForObjectLiteral`.